src/rsa.rs

```diff
@@ -345,14 +345,12 @@ impl RSAKeyPair {
         try!(padding_alg.pad(msg, signature));
         let mut rand = rand::RAND::new(rng);
         bssl::map_result(unsafe {
-            let blinding = BN_BLINDING_new();
-            let ret =
-                GFp_rsa_private_transform(self.rsa.as_ref(),
-                                          signature.as_mut_ptr(),
-                                          signature.len(), blinding,
-                                          &mut rand);
-            BN_BLINDING_free(blinding);
-            ret
+            let blinding: *mut BN_BLINDING = *(self.blinding.lock().unwrap());
```

Comment:
Ah, it looks like those annotations are now vestigial! In some previous iteration they were necessary to deref through the `MutexGuard`. Removing throughout.